samcli/commands/pipeline/bootstrap/cli.py

```diff
@@ -0,0 +1,221 @@
+        cloudformation_execution_role_arn=cloudformation_execution_role,
+        artifacts_bucket_arn=artifacts_bucket,
+        create_ecr_repo=create_ecr_repo,
+        ecr_repo_arn=ecr_repo,
+        pipeline_ip_range=pipeline_ip_range,
+        confirm_changeset=confirm_changeset,
+        config_file=config_env,
+        config_env=config_file,
+    )  # pragma: no cover
+
+
+def do_cli(
+    region: Optional[str],
+    profile: Optional[str],
+    interactive: bool,
+    stage_name: Optional[str],
+    pipeline_user_arn: Optional[str],
+    pipeline_execution_role_arn: Optional[str],
+    cloudformation_execution_role_arn: Optional[str],
+    artifacts_bucket_arn: Optional[str],
+    create_ecr_repo: bool,
+    ecr_repo_arn: Optional[str],
+    pipeline_ip_range: Optional[str],
+    confirm_changeset: bool,
+    config_file: Optional[str],
+    config_env: Optional[str],
+) -> None:
+    """
+    implementation of `sam pipeline bootstrap` command
+    """
+    if not pipeline_user_arn:
+        pipeline_user_arn = _load_saved_pipeline_user()
+
+    if interactive:
+        guided_context = GuidedContext(
+            stage_name=stage_name,
+            pipeline_user_arn=pipeline_user_arn,
+            pipeline_execution_role_arn=pipeline_execution_role_arn,
+            cloudformation_execution_role_arn=cloudformation_execution_role_arn,
+            artifacts_bucket_arn=artifacts_bucket_arn,
+            create_ecr_repo=create_ecr_repo,
+            ecr_repo_arn=ecr_repo_arn,
+            pipeline_ip_range=pipeline_ip_range,
+        )
+        guided_context.run()
+        stage_name = guided_context.stage_name
+        pipeline_user_arn = guided_context.pipeline_user_arn
+        pipeline_execution_role_arn = guided_context.pipeline_execution_role_arn
+        pipeline_ip_range = guided_context.pipeline_ip_range
+        cloudformation_execution_role_arn = guided_context.cloudformation_execution_role_arn
+        artifacts_bucket_arn = guided_context.artifacts_bucket_arn
+        create_ecr_repo = guided_context.create_ecr_repo
+        ecr_repo_arn = guided_context.ecr_repo_arn
+
+    if not stage_name:
+        raise click.UsageError("Missing required parameter '--stage-name'")
+
+    stage: Stage = Stage(
+        name=stage_name,
+        aws_profile=profile,
+        aws_region=region,
+        pipeline_user_arn=pipeline_user_arn,
+        pipeline_execution_role_arn=pipeline_execution_role_arn,
+        pipeline_ip_range=pipeline_ip_range,
+        cloudformation_execution_role_arn=cloudformation_execution_role_arn,
+        artifacts_bucket_arn=artifacts_bucket_arn,
+        create_ecr_repo=create_ecr_repo,
+        ecr_repo_arn=ecr_repo_arn,
+    )
+
+    stage.bootstrap(confirm_changeset=confirm_changeset)
+
+    stage.print_resources_summary()
+
+    try:
+        stage.save_config(
+            config_dir=PIPELINE_CONFIG_DIR, filename=PIPELINE_CONFIG_FILENAME, cmd_names=_get_command_name()
+        )
+    except Exception:
```

Comment:
What are the possible exception types here?

Reply:
The intention is to capture all kind of exceptions(including unexpected ones); At this point the resources are already bootstrapped, we don't prompt/confirm with the users that we are going to save these configs, i.e. they are unaware of that. So, the bootstrap command has already succeeded and the users should face any failure messages.